src/ucp/core/ucp_request.inl

```diff
@@ -788,4 +778,40 @@ ucp_request_invoke_uct_completion(ucp_request_t *req, ucs_status_t status)
     ucp_invoke_uct_completion(&req->send.state.uct_comp, status);
 }
 
+static UCS_F_ALWAYS_INLINE void
+ucp_request_complete_send(ucp_request_t *req, ucs_status_t status)
+{
+    ucs_trace_req("completing send request %p (%p) "UCP_REQUEST_FLAGS_FMT" %s",
+                  req, req + 1, UCP_REQUEST_FLAGS_ARG(req->flags),
+                  ucs_status_string(status));
+    UCS_PROFILE_REQUEST_EVENT(req, "complete_send", status);
+    ucp_request_complete(req, send.cb, status, req->user_data);
+}
+
+static UCS_F_ALWAYS_INLINE void
+ucp_request_complete_send_common(ucp_request_t *sreq, ucs_status_t status)
+{
+    ucp_request_send_generic_dt_finish(sreq);
+    ucp_request_send_buffer_dereg(sreq);
+    ucp_request_complete_send(sreq, status);
+}
+
+static UCS_F_ALWAYS_INLINE void
+ucp_request_complete_recv(ucp_request_t *req, ucs_status_t status)
```

Comment:
this is relevant for rndv only, right? I'd keep `rndv` in the name

Reply:
1) where is it used for eager?
2) If it is used, `UCP_REQUEST_FLAG_RECV_AM` has no meaning for eager, better avoid extra branch